### ISSUE TITLE
terminate called after throwing an instance of 'std::bad_weak_ptr'

### ISSUE THREAD

**Issue by kare** [paullouisageneau/libdatachannel#645]
While running Streamr broker node libdatachannel logs:
```
terminate called after throwing an instance of 'std::bad_weak_ptr'
  what():  bad_weak_ptr
```
And causes broker to crash. We have tried to reproduce the issue locally, but without success so far.
It would seem that the nodes running in sparse network (Asia) run into this issue regularly, but nodes in dense network (Europe) don't necessarily at all. So best guess is that high latency connections are possibly causing a race condition.

**Comment by paullouisageneau** [comment]
I think the only thing that may throw a `std::bad_weak_ptr` exception in the library codebase is `shared_from_this()`, and the only call to `shared_from_this()` from a callback missing a try-catch block is in `impl::SctpTransport::handleUpcall()`: https://github.com/paullouisageneau/libdatachannel/blob/33bc1b8922c6b577830e5938ec4ad210d80435b7/src/impl/sctptransport.cpp#L717

Indeed, usrsctp may trigger callbacks after socket close, and while there is a mechanism which prevents calls after destruction, it does not prevent calls during destruction which would lead to this exception.

**Comment by paullouisageneau** [comment]
Fix released in v0.17.6
